db/hardware_test.go

```diff
@@ -211,6 +211,118 @@ func TestDeleteHardware(t *testing.T) {
 	}
 }
 
+func TestGetByID(t *testing.T) {
+	ctx := context.Background()
+	tests := []struct {
+		// Name identifies the single test in a table test scenario
+		Name string
+		// InputAsync if set to true inserts all the input concurrently
+		InputAsync bool
+		// Input is a hardware that will be used to pre-populate the database
+		Input []*hardware.Hardware
+	}{
+		{
+			Name:  "get-hardware-by-id",
+			Input: []*hardware.Hardware{readHardwareData("./testdata/hardware.json")},
+		},
+		{
+			Name:       "stress-get-hardware-by-id",
+			InputAsync: true,
+			Input: func() []*hardware.Hardware {
+				input := []*hardware.Hardware{}
+				for ii := 0; ii < 10; ii++ {
+					hw := readHardwareData("./testdata/hardware.json")
+					hw.Id = uuid.New().String()
+					hw.Network.Interfaces[0].Dhcp.Mac = strings.Replace(hw.Network.Interfaces[0].Dhcp.Mac, "00", fmt.Sprintf("0%d", ii), 1)
+				}
+				return input
+			}(),
+		},
+	}
+
+	for _, s := range tests {
+		t.Run(s.Name, func(t *testing.T) {
+			t.Parallel()
+			_, tinkDB, cl := NewPostgresDatabaseClient(t, ctx, NewPostgresDatabaseRequest{
+				ApplyMigration: true,
+			})
+			defer func() {
+				err := cl()
+				if err != nil {
+					t.Error(err)
+				}
+			}()
+
+			for _, hw := range s.Input {
+				err := createHardware(ctx, tinkDB, hw)
+				if err != nil {
+					t.Error(err)
+				}
+			}
+
+			var wg sync.WaitGroup
+			wg.Add(len(s.Input))
+			for _, hw := range s.Input {
+				if s.InputAsync {
+					go func(ctx context.Context, tinkDB *db.TinkDB, hw *hardware.Hardware) {
+						defer wg.Done()
+						data, err := tinkDB.GetByID(ctx, hw.Id)
+						if err != nil {
+							t.Error(err)
+						}
+						h := &hardware.Hardware{}
+						if err := json.Unmarshal([]byte(data), h); err != nil {
+							t.Error(err)
+						}
+						if dif := cmp.Diff(hw, h, cmp.Comparer(hardwareComparer)); dif != "" {
+							t.Errorf(dif)
+						}
+					}(ctx, tinkDB, hw)
+				} else {
+					wg.Done()
```

Comment:
Overall I don't understand what you are doing. You should check what you expect as part of the `table` test. Not here